docs/source/_notebooks/xgboost-titanic.rst

```diff
@@ -0,0 +1,2608 @@
+            -0.060
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            SibSp
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                0.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 94.81%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.063
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            &lt;BIAS&gt;
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                1.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 92.71%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.102
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            Age
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                19.000
+            </td>
+        
+    </tr>
+            
+                <tr style="background-color: hsl(0, 100.00%, 83.30%); border: none;">
+        <td style="padding: 0 1em 0 0.5em; text-align: right; border: none;">
+            -0.333
+        </td>
+        <td style="padding: 0 0.5em 0 0.5em; text-align: left; border: none;">
+            Pclass=3
+        </td>
+        
+            <td style="padding: 0 0.5em 0 1em; text-align: right; border: none;">
+                1.000
+            </td>
+        
+    </tr>
+            
+    
+            </tbody>
+        </table>
+    
+                
+            
+    
+            
+    
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+        
+    
+    
+    
+
+
+
+
+5. Adding text features
+-----------------------
+
+Right now we treat *Name* field as categorical, like other text
+features. But in this dataset each name is unique, so XGBoost does not
+use this feature at all, because it's such a poor discriminator: it's
+absent from the weights table in section 3.
+
+But *Name* still might contain some useful information. We don't want to
+guess how to best pre-process it and what features to extract, so let's
+use the most general character ngram vectorizer:
+
+.. code:: python
+
+    from sklearn.pipeline import FeatureUnion
+    from sklearn.feature_extraction.text import CountVectorizer
+    
+    vec2 = FeatureUnion([
+        ('Name', CountVectorizer(
+            analyzer='char_wb',
+            ngram_range=(3, 4),
```

Comment:
Yes, that's reasonable. I just tried (2, 4): the quality is slightly less (although it means nothing: 0.822 ± 0.033 for (2, 4) vs 0.831 ± 0.006 for (3, 4)), it picks more features inside names, but "Mr" is handled by 3 grams too, because it's written as 3 letters: "Mr.". I also tried training a model on names alonee, it's again slightly better with (3, 4) than with (2, 4): 0.776 ± 0.039 for (2, 4) vs 0.788 ± 0.029 for (3, 4).
I think it's too small dataset to reliably check which is better, but predictions from (3, 4) model look a bit better in highlighted text, so I'd rather go with it.